jobapi/server.go

```diff
@@ -0,0 +1,108 @@
+package jobapi
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/buildkite/agent/v3/env"
+)
+
+type Server struct {
+	SocketPath string
+	environ    *env.Environment
```

Comment:
[All maps in go are reference types](https://go.dev/blog/maps), and are implicitly pointers anyway, butttttt i do kinda like the `*` in the struct (and the constructor below) to call out that "hello, this is a mutable thing and we're going to be mutating it", which might be slightly less clear without it?

on the other hand, it might just be confusing and say "man, why did that benno person do that" to future travellers.

Reply:
Suggestion: rewrite the whole thing in rust for those sweet sweet mutability markers
```Rust
struct Server {
  socket_path: String,
  environ: &mut HashMap<String, String>,
  token: String,
  started: bool,
}
```

Reply:
Yeah, IIRC there was something of a discussion pre-Go 1.0 about whether maps ought to carry the * to signify they're all references. Since that discussion is ancient history at this point, and it's a similar situation if it were an interface type, and we're not rewriting in Rust (yet), I recommend avoiding the *.